app/controllers/notes_controller.rb

```diff
@@ -61,7 +61,41 @@ def show
       @tags = @node.tags
       @tagnames = @tags.collect(&:name)
       @preview = false
-
+      @react = params[:react]
+
+      if params[:react]
+        # query everything we need in the comments state object
+        comments_record = @node
+          .comments_viewable_by(current_user)
+          .includes([:replied_comments, :node])
+          .order('timestamp ASC')
+        
+        comments = helpers.get_react_comments(comments_record)
+
+        currentUser = {
+          :canModerate => current_user.can_moderate?,
+          :id => current_user[:id],
+          :role => current_user[:role],
+          :status => current_user[:status]
+        }
+
+        @react_props = {
```

Comment:
Would this be better off inside the Comment model? Just an idea, maybe not though.

Reply:
> Would this be better off inside the Comment model? Just an idea, maybe not though.

Maybe?? This is one of those things I'm not familiar with concerning Rails. From a performance standpoint, a method in the `Comment` model to `get_initial_react_props` would be faster than what I'm doing in the controller right now&mdash;right? I can add it to the list of optimizations.